---
title: "RNA-Seq Analysis Mini Project"
author: "Kaitlyn Madriaga, A17217752"
format: pdf
editor: visual
---

## Differential Expression Analysis
```{r}
#| message: false
library(DESeq2)
```

Load data files
```{r}
metaFile <- "GSE37704_metadata.csv"
countFile <- "GSE37704_featurecounts.csv"

# Import metadata and take a peak
colData = read.csv(metaFile, row.names=1)
head(colData)
```

```{r}
# Import countdata
countData = read.csv(countFile, row.names=1)
head(countData)
```

> Q. Remove the first column ("length") from countData

```{r}
# Note we need to remove the odd first $length col
countData <- as.matrix(countData[,-1])
head(countData)
```

```{r}
# Filter count data where you have 0 read count across all samples.
countData = countData[rowSums(countData) > 0,]
#can also use [!(rowSums(countData) == 0), ]
head(countData)
```

How many genes are left? 
```{r}
nrow(countData)
```

## DESeq2

Set up formats and run analysis 

```{r}
dds = DESeqDataSetFromMatrix(countData=countData,
                             colData=colData,
                             design=~condition)
#condition is the name in the colData 

dds = DESeq(dds) 
```

get results from dds using `results()`

```{r}
res <- results(dds) 
head(res)
```

> Q. Call the summary() function on your results to get a sense of how many genes are up or down-regulated at the default 0.1 p-value cutoff.

```{r}
summary(res)
```

Plot results: 
```{r}
plot(res$log2FoldChange,-log(res$padj))
```
```{r}
# Make a color vector for all genes
mycols <- rep("gray", nrow(res) )

# Color red the genes with absolute fold change above 2
mycols[ abs(res$log2FoldChange) > 2 ] <- "red"

# Color blue those with adjusted p-value less than 0.01
#  and absolute fold change more than 2
inds <- (res$padj < 0.01) & (abs(res$log2FoldChange) > 2 )
mycols[ inds ] <- "blue"

plot( res$log2FoldChange, -log(res$padj), col=mycols, xlab="Log2(FoldChange)", ylab="-Log(P-value)" )
```

## Annotate Results 

I need to add annotation to my results including gene symbols and entrezeids, etc.  For this I will use the **AnnotationDbi** package. 

```{r}
library(AnnotationDbi)
library(org.Hs.eg.db)
```

```{r}
columns(org.Hs.eg.db)
```

```{r}
res$symbol <- mapIds(org.Hs.eg.db,
                     keys = row.names(res),
                     keytype = "ENSEMBL", 
                     column = "SYMBOL", 
                     multiVals = "first")

res$entrez <- mapIds(org.Hs.eg.db,
                     keys = row.names(res),
                     keytype = "ENSEMBL", 
                     column = "ENTREZID", 
                     multiVals = "first")

res$name <- mapIds(org.Hs.eg.db,
                     keys = row.names(res),
                     keytype = "ENSEMBL", 
                     column = "GENENAME", 
                     multiVals = "first")
```
```{r}
head(res)
```

Order results by adjusted p-value and save them to a CSV file 

```{r}
res = res[order(res$pvalue),]
write.csv(res, file="deseq_results.csv")
```

## Pathway Analysis 

We can use the KEGG database to get more insight 

```{r}
#| message: false
library(pathview)
library(gage)
library(gageData)

data(kegg.sets.hs)
data(sigmet.idx.hs)
```


The `sigmet.idx.hs` is an index of numbers of signaling and metabolic pathways in `kegg.set.gs`. 
We can examine them here: 

```{r}
# Focus on signaling and metabolic pathways only
kegg.sets.hs = kegg.sets.hs[sigmet.idx.hs]

# Examine the first 3 pathways
head(kegg.sets.hs, 3)
```

Make a new vector of the fold-change rules that I will use as input for gage, this will have the ENTREZ ID as names
```{r}
foldchanges = res$log2FoldChange
names(foldchanges) = res$entrez
head(foldchanges)
```

```{r}
# Get the results
keggres = gage(foldchanges, gsets=kegg.sets.hs)
```

```{r}
attributes(keggres)
```
Look at the top 3 "less"

```{r}
head(keggres$less)
```

Make a pathway plot: 

```{r}
pathview(gene.data=foldchanges, pathway.id="hsa04110")
```

![Cell Cycle Pathway Analysis](hsa04110.pathview.png)

```{r}
# A different PDF based output of the same data
#pathview(gene.data=foldchanges, pathway.id="hsa04110", kegg.native=FALSE)
```
> Q. Can you plot the pathview figures for the top 5 down-reguled pathways?

```{r}
keggrespathways <- rownames(keggres$less)[1:5]

# Extract the 8 character long IDs part of each string
keggresids = substr(keggrespathways, start=1, stop=8)
keggresids
```
```{r}
pathview(gene.data=foldchanges, pathway.id=keggresids, species="hsa")
```
![DNA Replication Pathway Analysis](hsa03030.pathview.png)
![RNA Transport Pathway Analysis](hsa03013.pathview.png)
![Homologous Recombination Pathway Analysis](hsa03440.pathview.png)
![Oocyte Meiosis Pathway Analysis](hsa04114.pathview.png)

